rosetta/retriever/retriever.go

```diff
@@ -174,84 +176,62 @@ func (r *Retriever) Block(rosBlockID identifier.Block) (*object.Block, []identif
 	}
 
 	// Then, get the header; it contains the block ID, parent ID and timestamp.
-	header, err := r.index.Header(*completed.Index)
+	header, err := r.index.Header(height)
 	if err != nil {
 		return nil, nil, fmt.Errorf("could not get header: %w", err)
 	}
 
 	// Next, we get all the events for the block to extract deposit and withdrawal events.
-	events, err := r.index.Events(*completed.Index, flow.EventType(deposit), flow.EventType(withdrawal))
+	events, err := r.index.Events(height, flow.EventType(deposit), flow.EventType(withdrawal))
 	if err != nil {
 		return nil, nil, fmt.Errorf("could not get events: %w", err)
 	}
 
-	// Convert events to operations and group them by transaction ID.
-	buckets := make(map[string][]object.Operation)
-	for _, event := range events {
-		op, err := r.convert.EventToOperation(event)
-		if errors.Is(err, converter.ErrIrrelevant) {
-			continue
-		}
-		if err != nil {
-			return nil, nil, fmt.Errorf("could not convert event: %w", err)
-		}
-
-		tID := event.TransactionID.String()
-		buckets[tID] = append(buckets[tID], *op)
+	// Get all transaction IDs for this height.
+	txIDs, err := r.index.TransactionsByHeight(height)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get transaction by height: %w", err)
 	}
 
-	// Iterate over all transactionIDs to create transactions with all relevant operations.
+	// Go over all the transaction IDs and create the related Rosetta transaction
+	// until we hit the limit, at which point we just add the identifier.
 	var blockTransactions []*object.Transaction
 	var extraTransactions []identifier.Transaction
-	var count int
-	for txID, operations := range buckets {
-		if count >= int(r.cfg.TransactionLimit) {
-			extraTransactions = append(extraTransactions, identifier.Transaction{Hash: txID})
+	for index, txID := range txIDs {
+		if index > int(r.cfg.TransactionLimit) {
+			extraTransactions = append(extraTransactions, identifier.Transaction{Hash: txID.String()})
```

Comment:
Could use the new `rosettaTxID` function:
```suggestion
			extraTransactions = append(extraTransactions, rosettaTxID(txID)
```